src/main/java/com/duke/bot/util/Parser.java

```diff
@@ -0,0 +1,115 @@
+package com.duke.bot.util; 
+
+import java.time.LocalDate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import com.duke.bot.DukeException;
+import com.duke.bot.command.AddCommand;
+import com.duke.bot.command.Command;
+import com.duke.bot.command.DeleteCommand;
+import com.duke.bot.command.DoneCommand;
+import com.duke.bot.command.ExitCommand;
+import com.duke.bot.command.FindCommand;
+import com.duke.bot.command.ListCommand;
+import com.duke.bot.task.Deadline;
+import com.duke.bot.task.Event;
+import com.duke.bot.task.Todo;
+
+public class Parser {
+    /**
+     * Parses a string input and return the command
+     * (one of the subclasses of {@link com.duke.bot.command.Command}) the input represents.
+     * 
+     * @param input String input to be parsed
+     * @return A command the string input represents
+     * @throws DukeException Throws if invalid input string is received
+     */
+    public static Command parse(String input) throws DukeException {
+        if (input.equalsIgnoreCase("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("done")) {
+            return parseDoneInput(input);
+        } else if (input.startsWith("delete")) {
+            return parseDeleteInput(input);
+        } else if (input.startsWith("find")) {
+            return parseFindInput(input);
+        } else if (input.startsWith("todo")) {
+            return parseTodoInput(input);
+        } else if (input.startsWith("event")) {
+            return parseEventInput(input);
+        } else if (input.startsWith("deadline")) {
+            return parseDeadlineInput(input);
+        } else if (input.equalsIgnoreCase("bye")) {
+            return new ExitCommand();
+        } else {
+            throw new DukeException(String.format("Oops, '%s' is an invalid input", input));
+        }
```

Comment:
Consider using `switch` and `case` statements instead of `if-else`, e.g.

    switch (input.split("\\s+")[0]) {
    case "list":
        return new ListCommand();
    case "done":
        return parseDoneInput(input);
    case "delete":
        return parseDeleteInput(input);
    // ...

Note that above will parse `list superfluous_argument` as a valid `ListCommand`. 

If you want commands for `list` and `bye` to explicitly not have any additional arguments passed in, that checking can be done in a `parseListInput` method.